github/search.go

```diff
@@ -71,6 +71,22 @@ func (s *SearchService) Repositories(ctx context.Context, query string, opt *Sea
 	return result, resp, err
 }
 
+// TopicsSearchResult represents the result of a topics search.
+type TopicsSearchResult struct {
+	Total             *int         `json:"total_count,omitempty"`
+	IncompleteResults *bool        `json:"incomplete_results,omitempty"`
+	Repositories      []Repository `json:"items,omitempty"`
```

Comment:
I don't think these results are `Repositories`.

In fact, to match the other `Search` endpoints, you will most likely need to create a new struct called `TopicResult` and then fill in the fields and tags from the GitHub v3 API developer docs.

So this should be something like:

```go
Topics []*TopicResult `json:items,omitempty`
```